Bottom: d29f68788329b814342fb7ed0b61fdf1f6397319
Top:    ebf67f3c90393d9456a53ba613d2a47e3a86a2cf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:34:57 +0000

Try debugging draw speed




---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index db7507a..4fae738 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -1810,13 +1810,12 @@ Benchmark (int argc, char **argv, int x, int y)
     {
       gdk_window_invalidate_rect (gport->drawing_area->window, NULL, 1);
       gdk_window_process_updates (gport->drawing_area->window, FALSE);
-      gdk_display_sync (display);
       time (&end);
       i++;
     }
   while (end - start < 10);
 
-  printf (_("%g redraws per second\n"), i / 10.0);
+  printf (_("%g redraws per second\n"), (double)i / (double)(end-start));
 
   return 0;
 }
